core/Pos/Binary/Class/Numbers.hs

```diff
@@ -204,156 +65,69 @@ newtype TinyVarInt = TinyVarInt {getTinyVarInt :: Word16}
-        FixedSizeInt <$> Store.peek -- CSL-1122 fix endianess
-    {-# INLINE get #-}
-    size = convertSize getFixedSizeInt Store.size
-    {-# INLINE size #-}
+  encode = encode . getFixedSizeInt
+  decode = FixedSizeInt <$> decode
 
 -- Word
 
 instance Bi (UnsignedVarInt Word) where
-    put (UnsignedVarInt a) = putUnsignedVarInt a
-    {-# INLINE put #-}
-    get = label "UnsignedVarInt Word" $
-        UnsignedVarInt <$> limitGet 15 getUnsignedVarInt'
-    {-# INLINE get #-}
-    size = VarSize $ getUnsignedVarIntSize . getUnsignedVarInt
-    {-# INLINE size #-}
+  encode = encode . getUnsignedVarInt
+  decode = UnsignedVarInt <$> decode
 
--- Is this instance valid at all? Is it word32 or word64 after all?
 instance Bi (FixedSizeInt Word) where
-    put (FixedSizeInt a) = Store.poke a -- CSL-1122 fix endianess
-    {-# INLINE put #-}
-    get = label "FixedSizeInt Word" $
-        FixedSizeInt <$> Store.peek -- CSL-1122 fix endianess
-    {-# INLINE get #-}
-    size = convertSize getFixedSizeInt Store.size
-    {-# INLINE size #-}
+  encode = encode . getFixedSizeInt
+  decode = FixedSizeInt <$> decode
 
 -- Word16
 
 instance Bi (UnsignedVarInt Word16) where
-    put (UnsignedVarInt a) = putUnsignedVarInt a
-    {-# INLINE put #-}
-    get = label "UnsignedVarInt Word16" $
-        UnsignedVarInt <$> limitGet 15 getUnsignedVarInt'
-    {-# INLINE get #-}
-    size = VarSize $ getUnsignedVarIntSize . getUnsignedVarInt
-    {-# INLINE size #-}
+  encode = encode . getUnsignedVarInt
+  decode = UnsignedVarInt <$> decode
 
 -- Word32
 
 instance Bi (UnsignedVarInt Word32) where
-    put (UnsignedVarInt a) = putUnsignedVarInt a
-    {-# INLINE put #-}
-    get = label "UnsignedVarInt Word32" $
-        UnsignedVarInt <$> limitGet 15 getUnsignedVarInt'
-    {-# INLINE get #-}
-    size = VarSize $ getUnsignedVarIntSize . getUnsignedVarInt
-    {-# INLINE size #-}
+  encode = encode . getUnsignedVarInt
+  decode = UnsignedVarInt <$> decode
 
 -- Word64
 
 instance Bi (UnsignedVarInt Word64) where
-    put (UnsignedVarInt a) = putUnsignedVarInt a
-    {-# INLINE put #-}
-    get = label "UnsignedVarInt Word64" $
-        UnsignedVarInt <$> limitGet 15 getUnsignedVarInt'
-    {-# INLINE get #-}
-    size = VarSize $ getUnsignedVarIntSize . getUnsignedVarInt
-    {-# INLINE size #-}
+  encode = encode . getUnsignedVarInt
+  decode = UnsignedVarInt <$> decode
 
 -- TinyVarInt
-
+-- The new CBOR instance is going to automatically offload on the `encodeWord16` from the CBOR library.
+-- The reasoning behind this is that the old implementation was taking 1 or 2 bytes according to the size,
+-- but due to the fact we need flat-term encoding for CBOR instances, we need (in case the TinyVarInt occupies
+-- two bytes) to store this as a list, so the space-saving would be lost anyway due to the list size, even
+-- in the case the `Int` occupies only 1 byte. Therefore, it's simpler to always encode this as a `Word16`
+-- directly.
```

Comment:
The comment should say it all, but just to summarise: In the new CBOR world it doesn't make a great deal of sense to "micromanage" the serialisation of these numeric types, as the storage saving is lost anyway due to the fact we would need to encode a list of `Word8` in case the `TinyVarInt` represents a number which requires 2 bytes, but then we will have to use an extra byte to store the length of the list, making the situation even worse. I am relatively new to CBOR, so I'm happy to review this with the team in the future, although (I guess), this shouldn't prevent this PR from being merged.